Add semantic release and a Github action to run it

jordancrane-truework opened the pull request:
This change uses npm to introduce semantic release and commitizen to enable easier releases in the
future.

API-46

jordancrane-truework committed:
ci: add semantic release and a Github action to run it

This change uses npm to introduce semantic release and commitizen to enable easier releases in the
future.

API-46

victorkabdebon reviewed (approved):
👍  Please get another set of eyes, but that looks fine to me.

estrattonbailey reviewed (approved):
All looks good to me afaict

jordancrane-truework committed:
ci: add dry run to CI, add husky to run commitizen

This change adds a release dry run to the CI action.
Additionally, it adds husky and a prepare script to
automatically configure git hooks for commitizen.
